Requesting and Notifying Out Of Office time (#233)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/OutOffOfficeRequest.md

1. Select Tools
1. Select Out of Office
1. ✔️ Send automatic replies for account `<your outlook account>`
1. Reply once to each sender with: `<Define a message mentioning the period you are on holidays>`
1. ✔️ Only send replies during this time period
1. Start time: First day of your holiday 
1. End time: Last day of your holiday

### Before leaving

When we leave on holidays we should make sure we leave our work either finished or delegated to someone else. There are a couple of steps that should be performed to reduce the impact of the absence.

1. Try to finish any work in progress before you go.
1. Resolve the reviews on your opened PRs and try to get them merged. 
1. In the event of not being able to complete 1. and/or 2. please sync with another iOS Engineer, ideally someone from your squad, and do a handover. When handing over an opened PR please make sure you add the iOS Engineer as `Assignee`.
1. Make sure your JIRA tickets are up-to-date. Any pending ticket should
	1. Have a description of what is left to be done.
	1. Who to contact if necessary.
	1. If it's something urgent should be assigned to another team member and a proper handover should be done with that person.

> 1. [Add yourself to the PullReminder exclusion list](#steps-to-be-added-to-the-pullreminder-exclusion-list) to prevent you to be added as a PR reviewer while you are away. 

Reviewer: beautiful 👌

Ideally, you should give at least 2 weeks notice before taking any holiday but there are exceptions for emergency situations. If you want to book more than 5 consecutive days you ideally should give 4 weeks notice.

### Steps to request

1. Ask for approval from your PM - only if approved you can proceed to the next steps.
2. Communicate it to your line manager and mention your PM approval
3. Request in Bamboo

### Steps to notify

1. [Send an event](#steps-to-configure-the-outlook-event) to the iOS Team Calendar and to your Squad email group
1. Update the [Team Plan document](https://docs.google.com/spreadsheets/d/1kdY3edy_TeqIGH_7VnZzElxgVo_qD2z4EF-arWNShyw/edit?usp=sharing)
1. Notify the iOS Team and your Squad via Slack on the working day before you go on holidays

If there is a big impact in the iOS Team or in your Squad due to you being away, please communicate a few days before in the team specific slack channel and do a handover if needed.

#### Steps to Configure the Outlook event
1. Open Outlook 
1. Select Calendar
1. Select Appointment

> 1. Select Invite

Reviewer 2: "Appointment button then Invite button" is the same as directly hitting the "Meeting" button, so it would be easier to collapse the two steps _(though TIL that one could easily convert an Appointment into a Meeting with that Invite button, very useful since double-click on a slot in the calendar opens the window to create an Appointment, not a Meeting, by default)_

1. To: `<iOS calendar email group>`, `<your squad email group>`
1. Subject: `<Your name>` Holidays 🌴
1. Duration: All-day event

> 1. Show as: Free

Reviewer 2: This has the benefit of people accidentally accepting your invite not being marked as Busy (as outlook would consider that accepting a meeting invite would make you part of it so busy at the time of appointment), but has the drawback of not marking the person actually being on holidays as Busy.

This means that other people wanting to later organise any meetings while that person is away won't see that person as "not available" in the Scheduling view.

Two possible suggestions to fix that:

1. Either keep that invite to communicate with the team as "Free" but then we need to also create an additional all-day Appointment event in our own calendar only, without inviting anybody, marking us as Busy. This would make us have two similar events in our calendars (one for the invite shared with the group as free, one only local marking us as busy) but would avoid other people accidentally "accepting" our shared "meeting" invite to be marked as busy themselves
2. Or we should select "Busy" and not "Free" in that invite we send to the team, but need to make sure to communicate that the team receiving that invite should NOT accept NOR reject the invite, but just delete the email. Deleting the email will neither accept the invite (preventing you to be marked as busy) nor reject it (preventing it from being removed from the calendar), it will just keep the event in the calendar without marking you busy. The benefit of this solution is having only one event, the drawback being that it's easy to accept by mistake / force of habit and be marked as busy.

Author: I can see that having a person busy in the calendar can be useful, although I don't feel that the options presented are feasible:
1. We would have to create 2 events and as you mentioned:
>This would make us have two similar events in our calendars (one for the invite shared with the group as free, one only local marking us as busy)
2. I feel this can be very prone to error and I'm afraid people start to accept/reject meetings by mistake.

So at the moment, if anyone invites to a meeting and one of the attendees is going to be on holiday, the attendee should reject the meeting and should mention why, including the days this person is on holidays. 
In the event that the invite is sent when the person is already on holidays, then maybe we can add in our process for everyone to add an automatic reply to their outlook stating the days that he/she is on holiday. This message would be sent every time someone tries to invite that person to a meeting. This would also be useful, not for meeting invites, but also for any other email sent to that person.

Reviewer 2: I agree that adding a point about setting up email auto-reply is a good idea.

Author: Great! I will update this document with that!

Author: Done 🙂

# Requesting and Notifying Out Of Office time

> Someone who is Out Of Office means that is not in their usual work place at their usual working hours. This can be when someone goes on holidays or when someone that usually works in the office is working from somewhere else.

Reviewer 3: ``
```suggestion
Someone who is Out Of Office means that they are not in their usual work place at their usual working hours. This can be when someone goes on holidays or when someone that usually works in the office is working from somewhere else.
```

# Requesting and Notifying Out Of Office time

Someone who is Out Of Office means that they are not in their usual work place at their usual working hours. This can be when someone goes on holidays or when someone that usually works in the office is working from somewhere else.

OOO time has to be requested and approved by your Line Manager and your Project Manager.

There are different types of OOO and each one of them have different requirements for requesting and for being notified.

## Holidays 🌴

> As per contract, we are entitled to 25 holiday days per year. You can book your holidays at your convience, you might just need to sync first with your Squad and the other iOS engineers to see if everybody is OK with it. 

Reviewer 4: ```suggestion
As per contract, we are entitled to 25 holiday days per year. You can book your holidays at your convenience, you might just need to sync first with your Squad and the other iOS engineers to see if everybody is OK with it. 
```

Also, with the other iOS engineers? All 30 of them?

Ideally, you should give at least 2 weeks notice before taking any holiday but there are exceptions for emergency situations. If you want to book more than 5 consecutive days you ideally should give 4 weeks notice.

### Steps to request

1. Ask for approval from your PM - only if approved you can proceed to the next steps.
2. Communicate it to your line manager and mention your PM approval

> 3. Request in Bamboo

Reviewer 4: If we're really asking the rest of the team for permission to go on holidays (see previous comment) then we're missing a step here

If there is a big impact in the iOS Team or in your Squad due to you being away, please communicate a few days before in the team specific slack channel and do a handover if needed.

#### Steps to Configure the Outlook event
1. Open Outlook 
1. Select Calendar
1. Select Meeting
1. To: `<iOS calendar email group>`, `<your squad email group>`
1. Subject: `<Your name>` Holidays 🌴
1. Duration: All-day event
1. Show as: Free
1. Reminder: None
1. Select Save & Close

#### Steps to Configure automatic reply
1. Open Outlook 
1. Select Tools
1. Select Out of Office
1. ✔️ Send automatic replies for account `<your outlook account>`
1. Reply once to each sender with: `<Define a message mentioning the period you are on holidays>`
1. ✔️ Only send replies during this time period
1. Start time: First day of your holiday 
1. End time: Last day of your holiday

### Before leaving

When we leave on holidays we should make sure we leave our work either finished or delegated to someone else. There are a couple of steps that should be performed to reduce the impact of the absence.

1. Try to finish any work in progress before you go.
1. Resolve the reviews on your opened PRs and try to get them merged.

> 1. In the event of not being able to complete 1. and/or 2. please sync with another iOS Engineer, ideally someone from your squad, and do a handover. When handing over an opened PR please make sure you add the iOS Engineer as `Assignee`.

Reviewer 2: 1. Also be sure that your tickets on JIRA are up-to-date, and especially mention any relevant info (like pending state, what's left to be done, who to contact about this particular ticket in your absence...) in the tickets that are not finished yet at the time you leave.

When working from home you should notify in the the iOS team Slack channel when you are away and when are you expecting to come back.

### Request

1. Ask for approval from your PM - only if approved you can proceed to the next steps.
2. Communicate it to your line manager and mention your PM approval
3. Request in Bamboo

### Notify

1. Notify the iOS Team and your Squad via Slack on the working day before you are working from home. Emergencies might happen that require you to work from home and only notify on the same day. That will be acceptable if there is a reasonable excuse for it.
2. On the day you are working from home change your Slack status to 🏡 **Working remotely**.

### Being remote

For the majority of the engineers in the team, the office is our usual place of work. We are used to work next to our colleagues and is natural that we end up communicating more verbally. In the office everyone in your Squad is aware of where you are and what you are doing, so there is no need for us to keep notifying each other.

> Working remotely requires a different mindset. We have to bear in mind that people in the office should still be aware of what is happening so you should:

Reviewer 2: tbh that should be true for non-remote as well, in the sense that non-remote people should be aware that their collegues are remote (some of us even are full-time remote). So that sentence above:

> We are used to work next to our colleagues and is natural that we end up communicating more verbally. In the office everyone in your Squad is aware of where you are and what you are doing, so there is no need for us to keep notifying each other.

is not really true and a bit annoying to read that it could be the mindset that people usually working in the office could have. Because in practice everybody should already be aware that a lot of people are working remote every day (either people used to work in office but WFH, or full-time remote) and even people in the office should communicate clearly via slack, to the remote people.

Author: That's true. I will re-phrase that. This was more directed to squads that have just non-remote people.

- Notify in Slack when your working day starts, finishes and any time you are off during the day. This should be written in the iOS-underground and in your Squad's Slack channels
- In addition to the iOS stand-up notes, you should communicate what are you planning to do during the day in your Squad's stand-up if you have one or in your Squad's Slack channel.

> 

Reviewer 2: There's a dedicated document with useful tips for remote workers, would be useful to link to it somewhere here

https://docs.google.com/document/d/1hjKtx1pPL_pxhLEuW-46w2XwxsC-mkw3Cp5uaHB2P8c/edit

Author: I will add that.
Thanks!

1. [Send an event](#steps-to-configure-the-outlook-event) to the iOS Team Calendar and to your Squad email group
1. Update the [Team Plan document](https://docs.google.com/spreadsheets/d/1kdY3edy_TeqIGH_7VnZzElxgVo_qD2z4EF-arWNShyw/edit?usp=sharing)
1. Notify the iOS Team and your Squad via Slack on the working day before you go on holidays.
1. [Enable automatic reply message](#steps-to-configure-automatic-reply) in your Babylon Outlook account.

> 1. At the end of your last working day before going on holidays change your Slack status to 🌴**On holiday** and set to clear after your last holiday day. 

Author: I think this is a good idea and I think we should apply this rule for anyone who goes on holidays for more than 1 week.

> 1. At the end of your last working day before going on holidays change your Slack status to 🌴**On holiday** and set to clear after your last holiday day. 

Author: I've done this. 😊

### Steps to notify

> 1. Notify the iOS Team and your Squad via Slack on the working day before you are working from home. Emergencies might happen that require you to work from home and only notify on the same day. That will be acceptable if there is a plausible reason for it.

Reviewer 2: I'm not a fan of the use of "plausible" here. Not sure if it's because I'm not native or not, but it feel to me like it has a connotation of "we will believe your excuse"... as if it was about lying to justify the WFH reason... Maybe use "an admissible reason" instead?

Author: ```suggestion
1. Notify the iOS Team and your Squad via Slack on the working day before you are working from home. Emergencies might happen that require you to work from home and only notify on the same day. That will be acceptable if there is a reasonable excuse for it.
```
